pkg/component/component.go

```diff
@@ -135,24 +149,81 @@ func GetCurrent(client *occlient.Client, applicationName string, projectName str
 	currentComponent := cfg.GetActiveComponent(applicationName, projectName)
 
 	return currentComponent, nil
-
 }
 
-// PushLocal start new build and push local dir as a source for local or a binary file for a binary build
+// PushLocal push local code to the cluster and trigger build there.
 // asFile indicates if it is a binary component or not
-func PushLocal(client *occlient.Client, componentName string, dir string, asFile bool) error {
-	err := client.StartBinaryBuild(componentName, dir, asFile)
+func PushLocal(client *occlient.Client, componentName string, applicationName string, path string, asFile bool, out io.Writer) error {
+	const targetPath = "/opt/app-root/src"
+
+	if !asFile {
+		// We need to make sure that thee is a '/' at the end, otherwise rsync will sync files to wrong directory
+		path = fmt.Sprintf("%s/", path)
+	}
+	// Find DeploymentConfig for component
+	componentLabels := componentlabels.GetLabels(componentName, applicationName, false)
+	componentSelector := util.ConvertLabelsToSelector(componentLabels)
+	dc, err := client.GetOneDeploymentConfigFromSelector(componentSelector)
+	if err != nil {
+		return errors.Wrap(err, "unable to get deployment for component")
+	}
+	// Find Pod for component
+	podSelector := fmt.Sprintf("deploymentconfig=%s", dc.Name)
+	// Wait for Pod to be in running state otherwise we can't sync data to it.
+	pod, err := client.WaitAndGetPod(podSelector)
+	if err != nil {
+		return errors.Wrapf(err, "error while waiting for pod  %s", podSelector)
+	}
+	var syncOutput string
+	if !asFile {
+		syncOutput, err = client.RsyncPath(path, pod.Name, targetPath)
+	} else {
+		syncOutput, err = client.CopyFile(path, pod.Name, targetPath)
+
+	}
 	if err != nil {
-		return errors.Wrap(err, "unable to start build")
+		return errors.Wrap(err, "unable push files to pod")
 	}
+	fmt.Fprintf(out, syncOutput)
+	fmt.Fprintf(out, "Please wait, building component....\n")
+
+	pipeReader, pipeWriter := io.Pipe()
+	go func() {
+		color.Set(color.FgYellow)
+		io.Copy(out, pipeReader)
+		defer color.Unset()
```

Comment:
It looks like you're using this within a `func()` and it's not being set correctly since the Go routine could exit at any time... 

From what I recall, the actual `color.Set` modifies the actual OS terminal rather than the Go binary output. What's happening is that the colour is being set to yellow in the terminal, but it's not being *unset* since the rest of the program is continuing while this Goroutine is running in the background..

Example:

1. PushLocal function starts
2. Goes into the Go routine
3. Colour is set to yellow
4. Rest of the program continues in Yellow
?????? (when the Go routine ends). Finally unset the colour

@kadel